### PR TITLE
[mkbundle] Enhanced mkbundle cygwin/mingw fallback support.

### PR DESCRIPTION
**Pull request by lateralusX**
By default, mkbundle on Windows will try to locate and use Visual Studio toolchain. Currently there is however no matching mono distribution including latest pre-build libraries for --static use case and if mono is just build locally using make, build won't automatically compile the VS solution and needed library. The correct steps are documented on http://www.mono-project.com/docs/compiling-mono/windows/, but if just make was used for local build, local install will only include mingw build libraries. This commit simplifies that scenario making it easier to use local mingw build together with
mkbundle.

NOTE, export MONOPREFIX poiting to mono install directory to make sure mkbundle will pick local mono mingw build.

Commit also includes a fallback in case where VS2015/VS2017 is used but Clang with
Microsoft CodeGen has not been installed. In such case mkbundle will fallback to mingw as.exe.

### CODE REVIEW

File: mcs/tools/mkbundle/mkbundle.cs [hunk 1]
```diff
@@ -2291,25 +2315,54 @@ static void AddSystemLibraries (ToolchainProgram program, bool staticLinkMono, b
...
+	static string GetMonoLibraryPath (ToolchainProgram program, bool staticLinkMono, bool staticLinkCRuntime)
+	{
+		string monoLibraryDir = Path.Combine (GetMonoDir (), "lib");
+		string monoLibrary = GetMonoLibraryName (program, staticLinkMono, staticLinkCRuntime);
+
+		if (Path.IsPathRooted (monoLibrary))
+			return monoLibrary;
+
+		if (program.IsVSToolChain) {
+			if (!monoLibrary.EndsWith (".lib"))
```
Comment (marek-safar) [inline mcs/tools/mkbundle/mkbundle.cs]: Should use ordinary comparer

File: mcs/tools/mkbundle/mkbundle.cs [hunk 2]
```diff
@@ -2291,25 +2315,54 @@ static void AddSystemLibraries (ToolchainProgram program, bool staticLinkMono, b
...
+				monoLibrary = monoLibrary + ".lib";
+		} else {
+			if (!monoLibrary.StartsWith ("lib"))
```
Comment (marek-safar) [inline mcs/tools/mkbundle/mkbundle.cs]: Should use ordinary comparer

File: mcs/tools/mkbundle/mkbundle.cs [hunk 3]
```diff
@@ -2291,25 +2315,54 @@ static void AddSystemLibraries (ToolchainProgram program, bool staticLinkMono, b
...
-				if (program.IsGCCToolChain) {
-					Console.WriteLine (	@"Warning: Static linking using default Visual Studio build libmono-static-sgen" +
-								@"might cause link errors when using GCC toolchain.");
-				}
-				monoLibrary = "libmono-static-sgen";
+				if (!monoLibrary.EndsWith (".dll.a"))
+					monoLibrary = monoLibrary + ".dll.a";
 			} else {
-				monoLibrary = "mono-2.0-sgen";
+				if (!monoLibrary.EndsWith (".a"))
```
Comment (marek-safar) [inline mcs/tools/mkbundle/mkbundle.cs]: Should use ordinary comparer

**Comment by lateralusX** [comment]
@marek-safar good to go?

**Pull request merged**